vunit/test/unit/test_configuration.py

```diff
@@ -67,6 +67,14 @@ def test_adds_tb_path_generic(self, tempdir):
         config_tb_path = Configuration('name', design_unit_tb_path)
         self.assertEqual(config_tb_path.generics["tb_path"], (tb_path + "/").replace("\\", "/"))
 
+    def test_constructor_adds_no_attributes(self):
+        with _create_config() as config:
+            self.assertEqual({}, config.attributes)
+
+    def test_constructor_adds_supplied_attributes(self):
```

Comment:
Adding an attribute named `foo` should be illegal and generate an error. `.foo` is ok. Add check and test for this.